code/modules/reagents/grenade_launcher.dm

```diff
@@ -52,15 +57,47 @@
 	for(var/mob/O in viewers(world.view, user))
```

Comment:
While we are at it, can you remove this for loop and use visible_message instead?

Reply:
Show him how please